src/QMCWaveFunctions/Jastrow/BsplineFunctor.h

```diff
@@ -98,24 +95,28 @@ struct BsplineFunctor : public OptimizableFunctorBase
     DeltaR       = cutoff_radius / (real_type)(numKnots - 1);
     DeltaRInv    = 1.0 / DeltaR;
     Parameters.resize(n);
-    SplineCoefs.resize(numCoefs);
+    spline_coefs_ = std::make_shared<Vector<real_type, OffloadAllocator<value_type>>>(numCoefs);
```

Comment:
So this pointer is shared with clones of this BsplineFunctor?  I guess it would be copied by the default copy constructor.

Reply:
BsplineFunctor has internal states, so it should be unique_ptr.
spline_coefs_ is for the spline function parameters so it should be shared_ptr across all the per walker instances of BsplineFunctor.